File: app/src/main/java/de/timbolender/fefereader/service/BootServiceStarter.java
package de.timbolender.fefereader.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import java.util.Objects;

/**
 * Broadcast receiver starting the update service. Intended for automatic launching of service after boot.
 */
public class BootServiceStarter extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = Objects.requireNonNull(intent.getAction());
        if(!action.equals("android.intent.action.BOOT_COMPLETED") &&
           !action.equals("android.intent.action.MY_PACKAGE_REPLACED"))
            return;

        UpdateService.startManualUpdate(context);
    }
}

File: app/src/main/java/de/timbolender/fefereader/service/BootServiceStarter.kt
package de.timbolender.fefereader.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Broadcast receiver starting the notification service. Sets up automatic updates as well.
 */
class BootServiceStarter : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action!!
        if (action != "android.intent.action.BOOT_COMPLETED" && action != "android.intent.action.MY_PACKAGE_REPLACED")
            return

        NotificationService.startService(context)
    }
}
